src/parser.ts

```diff
@@ -0,0 +1,272 @@
+ * @param text string representing "true" or "false"
+ */
+export function textToBoolean(text: string): boolean {
+  const sanitizedText = text.trim().toLowerCase();
+  if (sanitizedText === 'true') {
+    return true;
+  } else if (sanitizedText === 'false') {
+    return false;
+  }
+
+  throw new Error(
+    `Valid values can only be true or false. Received: "${text}"`
+  );
+}
+
+const validFormats: string[] = [...VALID_BASE_FORMATS];
+
+/**
+ * Extracts the format wrapped in `list()` or throws an error if it's an invalid format
+ * @param format a format string wrapped with `list()`
+ */
+export function extractListFormat(format: string): BaseVariableFormat {
+  let listValue = removePrefix('list(', format);
+  listValue = listValue.substr(0, listValue.length - 1).trim();
+  if (validFormats.includes(listValue)) {
+    return listValue as BaseVariableFormat;
+  } else {
+    throw new Error(`Invalid list format value. Received "${listValue}"`);
+  }
+}
+
+/**
+ * Extracts the formats wrapped in `nested_list()` or throws an error if one is an invalid format
+ * @param format a format string wrapped with `nested_list()`
+ */
+export function extractNestedListFormats(
+  format: string
+): [BaseVariableFormat, BaseVariableFormat] {
+  let listValues = removePrefix('nested_list(', format);
+  listValues = listValues.substr(0, listValues.length - 1).trim();
+  const [listValueX, listValueY] = listValues.split(',').map(trim);
+  if (validFormats.includes(listValueX) && validFormats.includes(listValueY)) {
+    return [listValueX as BaseVariableFormat, listValueY as BaseVariableFormat];
+  } else {
+    throw new Error(
+      `Invalid nested list format value. Received "${listValueX}" and "${listValueY}"`
+    );
+  }
+}
+
+/**
+ * Parses string to check if it's a valid format. If it isn't it will throw an error. Otherwise it will return a sanitized version
+ * @param text string to validate as format
+ */
+export function textToFormat(text: string): VariableFormat {
+  const sanitizedText = text.trim().toLowerCase();
+  if (validFormats.includes(sanitizedText)) {
+    // regular base format
+    return sanitizedText;
+  } else if (sanitizedText.startsWith('list(') && sanitizedText.endsWith(')')) {
+    // list format
+    const listValue = extractListFormat(sanitizedText);
+    return `list(${listValue})`;
+  } else if (
+    sanitizedText.startsWith('nested_list(') &&
+    sanitizedText.endsWith(')')
+  ) {
+    // nested list format
+    const [listValueX, listValueY] = extractNestedListFormats(sanitizedText);
+    return `nested_list(${listValueX},${listValueY})`;
+  }
+
+  throw new Error(`Invalid format. Received: "${text}"`);
+}
+
+/**
+ * Removes a specified prefix from the beginning of a string and trims whitespace
+ * @param prefix text to be removed
+ * @param line text to remove the prefix from
+ */
+export function removePrefix(prefix: string, line: string): string {
+  return line.substr(prefix.length).trim();
```

Comment:
should this check to ensure the `prefix` exists on the `line` before calling `subst`?

Reply:
should it also attempt to handle the close `)` and ensure it throws if missing?

Reply:
Good point on checking for the prefix. I'm against looking for the closing one because this function is called generally during the parsing to remove parts of the string